fix computerFileSize to parse correctly, prerapation to fix #25358

individual-it opened the pull request:
## Description
I've added some more tests for computerFileSize and they would fail, so fixed
the function to be more picky on what input to accept and try to parse and also fixed the rounding of values without units

## Motivation and Context
@PVince81 mentioned this function in the context of fixing #25358 but it does not work correct
1. input of "B" would return "NaN" and not null
2. invalid values like "40/0" or "40,30 kb" would be parsed
3. negative values would be parsed and returned as negative values
4. values without unit would not be rounded to full bytes. 125.5 would return 125.5, but I don't think oC runs on 4bit systems :-)

That are exact the problems we have in #25358
After this PR is accepted I'm happy to look into fixing #25358

## How Has This Been Tested?
wrote more tests

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

individual-it committed:
fix computerFileSize to be mory picky

individual-it committed:
more tests for computerFileSize

individual-it committed:
codestyle fix

individual-it committed:
fix double initialization

PVince81 commented on core/js/tests/specs/coreSpec.js:
```diff
@@ -605,7 +611,14 @@ describe('Core base tests', function() {
 					['116.4 TB', 127983153473126],
-					['116.4tb', 127983153473126]
+					['116.4tb', 127983153473126],
+					['8776656778888777655.4tb', 9.650036181387265e+30],
```
Looks like we're already ready for the next century's storage requirements 😆

individual-it replied:
living in the foothills of the Himalayas I know about huge clouds, so better be prepared, who knows what the climate change will bring ;-)

PVince81 commented on core/js/js.js:
```diff
@@ -1770,12 +1766,18 @@ OC.Util = {
-		var matches = s.match(/([kmgtp]?b?)$/i);
-		if (matches[1]) {
-			bytes = bytes * bytesArray[matches[1]];
+		var matches = s.match(/^[\s+]?([0-9]*)(\.([0-9]+))?( +)?([kmgtp]?b?)$/i);
```
I suggest to pre-create the regexp and store it in a hidden variable `this._computerFileSizeRegexp` or so.
I'm not sure if modern Javascript engines recompile the Regexp every time or whether it already caches it somewhere.

To be safe, store it somewhere so if this function is called in a loop it will run a bit faster.

individual-it replied:
changed it. please check it

individual-it committed:
pre-compile and store regexp in a hidden variable

as suggested by @PVince81 in Review https://github.com/owncloud/core/pull/27257#discussion_r103159658

PVince81 commented:
Looks better 👍

individual-it committed:
test strings with whitespaces

individual-it committed:
trim string before parsing

individual-it commented:
thought that a user might enter values with white spaces, and that should be possible and valid

PVince81 commented:
Oh, screw Jenkins...

I ran the JS tests locally and they pass.
